dissect/target/helpers/ssh.py

```diff
@@ -0,0 +1,187 @@
+            self._parse_pem(data)
+
+        else:
+            raise ValueError
+
+    def _parse_rfc4716(self, data: bytes) -> None:
+        """
+        Parse OpenSSH format SSH private keys.
+
+        The format:
+        "openssh-key-v1"0x00    # NULL-terminated "Auth Magic" string
+        32-bit length, "none"   # ciphername length and string
+        32-bit length, "none"   # kdfname length and string
+        32-bit length, nil      # kdf (0 length, no kdf)
+        32-bit 0x01             # number of keys, hard-coded to 1 (no length)
+        32-bit length, sshpub   # public key in ssh format
+            32-bit length, keytype
+            32-bit length, pub0
+            32-bit length, pub1
+        32-bit length for rnd+prv+comment+pad
+            64-bit dummy checksum?  # a random 32-bit int, repeated
+            32-bit length, keytype  # the private key (including public)
+            32-bit length, pub0     # Public Key parts
+            32-bit length, pub1
+            32-bit length, prv0     # Private Key parts
+            ...                     # (number varies by type)
+            32-bit length, comment  # comment string
+            padding bytes 0x010203  # pad to blocksize (see notes below)
+
+        Source: https://coolaj86.com/articles/the-openssh-private-key-format/
+        """
+
+        key_data = self._decode_rfc4716(data)
+
+        ssh_private_key_parser = cstruct.cstruct(endian=">")
+        ssh_private_key_parser.load(self.RFC4716_STRUCT)
+        private_key = ssh_private_key_parser.ssh_private_key(key_data)
+
+        # RFC4716 only support 1 key at the moment.
+        if private_key.magic != self.RFC4716_MAGIC or private_key.number_of_keys != 1:
+            raise ValueError
+
+        self.is_encrypted = private_key.cipher.value != self.RFC4716_NONE
+
+        self.public_key = base64.b64encode(private_key.public.value)
+        public_key_type = ssh_private_key_parser.ssh_string(private_key.public.value)
+        self.key_type = public_key_type.value
+
+        if not self.is_encrypted:
+            private_key_data = private_key.private.value.rstrip(self.RFC4716_PADDING)
+
+            # We skip the two dummy uint32s at the start.
+            private_key_index = 8
+
+            private_key_type = ssh_private_key_parser.ssh_string(private_key_data[private_key_index:])
+            private_key_index += 4 + private_key_type.length
+            self.key_type = private_key_type.value
+
+            private_key_fields = []
+            while private_key_index < len(private_key_data):
+                field = ssh_private_key_parser.ssh_string(private_key_data[private_key_index:])
+                private_key_index += 4 + field.length
+                private_key_fields.append(field)
+
+            # There is always a comment present (with a length field of 0 for empty comments).
+            self.comment = private_key_fields[-1].value
+
+    @staticmethod
+    def _validate_rfc4716(data) -> bool:
```

Comment:
Can you move all of these `_validate_*` methods into the global scope, and rename them `is_*`? Also this one in particular is missing the type hint for `data`.